<?xml version="1.0"?>
<launch>  
  <param name="/use_sim_time" value="true"/>

  <arg name="map_file" default="$(find turtlebot_gazebo)/maps/playground.yaml"/>  
  <param name="starting_location" value="0,0"/>
  <param name="goal_location" value="-3,3"/>
  <param name="robot_size" value="0.35"/>
  <param name="map_resolution" value="0.05"/>
  
  <param name="linear_speed" value="1.2" /> <!-- set it for the bot moving forward-->
  <param name="rotate_speed" value="0.96" /> <!-- set it for the bot rotate -->
  <param name="distance_threshold" value="0.01" /> <!-- set it for close enough points -->
  <param name="angular_threshold" value="0.01" /> <!-- set it for close enough direction -->

  <!-- Load the map -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" /> 
  
  <!--  Launch turtle bot world  -->
  <include file="$(find turtlebot_gazebo)/launch/turtlebot_world.launch"/>

  <!-- Launch amcl -->
  <node pkg="amcl" type="amcl" name="amcl" respawn="true">
	<param name="initial_pose_x" value="0" />
	<param name="initial_pose_y" value="0" />
	<param name="initial_pose_a" value="0" />

	<param name="odom_model_type" value="omni"/>
	<param name="transform_tolerance" value="0.3" />
	<param name="gui_publish_rate" value="0.1"/>
	<param name="laser_max_beams" value="60"/>
	<param name="min_particles" value="250"/>
	<param name="kld_err" value="0.01"/>
	<param name="kld_z" value="0.99"/>
	<param name="odom_alpha1" value="0.2"/>
	<param name="odom_alpha2" value="0.2"/>
	<param name="odom_alpha3" value="0.8"/>
	<param name="odom_alpha4" value="0.2"/>

	<param name="odom_alpha5" value="0.2"/>
	<param name="laser_z_hit" value="0.5"/>
	<param name="laser_z_short1" value="0.05"/>
	<param name="laser_z_max" value="1.05"/>
	<param name="laser_z_rand" value="0.5"/>
	<param name="laser_sigma_hit" value="0.2"/>
	<param name="laser_lambda_short" value="0.1"/>
	<param name="laser_lambda_short" value="0.1"/>
	<param name="laser_model_type" value="likelihood_field"/>
	<param name="laser_likelihood_max_dist" value="2.0"/>
	<param name="update_min_d" value="0.2"/>	
	<param name="update_min_a" value="0.3"/>	   
	<param name="resample_interval" value="1"/>
	<param name="transform_tolerance" value="0.1"/>
	<param name="recovery_alpha_slow" value="0.0"/>
	<param name="recovery_alpha_fast" value="0.0"/>
	<remap from="scan" to="/scan" />
	<remap from="map" to="/map" />
	<param name="map_frame_id" value="map"/>
	<param name="odom_frame_id" value="odom"/>
	<param name="base_frame_id" value="base_link"/>
	<param name="max_particles" value="1000"/>
  </node>

  <!-- Launch navigation_planner node  -->
  <node name="navigation_task" pkg="navigation_task" type="navigation_task.py" output="screen"/> 
</launch>

